❌ MongoDB connection error: MongoServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at Topology.selectServer (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/sdam/topology.ts:628:30)
    at async Topology._connect (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/sdam/topology.ts:468:22)
    at async Topology.connect (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/sdam/topology.ts:405:7)
    at async topologyConnect (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/mongo_client.ts:620:9)
    at async MongoClient._connect (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/mongo_client.ts:632:7)
    at async MongoClient.connect (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/mongo_client.ts:544:7)
    at async NativeConnection.createClient (/home/runner/workspace/node_modules/mongoose/lib/drivers/node-mongodb-native/connection.js:351:3)
    at async NativeConnection.openUri (/home/runner/workspace/node_modules/mongoose/lib/connection.js:1094:5)
    at async connectDB (/home/runner/workspace/server/db.ts:29:19)
    at async MongoStorage.createUser (/home/runner/workspace/server/storage.ts:69:7)
    at async Strategy._verify (/home/runner/workspace/server/auth.ts:92:25) {
  errorLabelSet: Set(0) {},
  reason: TopologyDescription {
    type: 'Unknown',
    servers: Map(1) { 'localhost:27017' => [ServerDescription] },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: null,
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined,
  [cause]: MongoNetworkError: connect ECONNREFUSED 127.0.0.1:27017
      at Socket.<anonymous> (/home/runner/workspace/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:394:16)
      at Object.onceWrapper (node:events:633:26)
      at Socket.emit (node:events:518:28)
      at emitErrorNT (node:internal/streams/destroy:169:8)
      at emitErrorCloseNT (node:internal/streams/destroy:128:3)
      at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
    errorLabelSet: Set(1) { 'ResetPool' },
    beforeHandshake: false,
    [cause]: Error: connect ECONNREFUSED 127.0.0.1:27017
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1611:16) {
      errno: -111,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 27017
    }
  }
}